### PR TITLE
WFP-322: added default pipeline

### PR DESCRIPTION
**Pull request by carlov20**

### CODE REVIEW

**Review by markreesmoj — CHANGES_REQUESTED** [review]
I think the build job needs the hmpps-common-vars context and the rest is nitpicking

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+          command: npm run lint
+      - persist_to_workspace:
+          root: .
+          paths:
+            - node_modules
+            - build-info.json
+            - build
+            - dist
+
+  check_outdated:
```
[the remark this answers is not in the document]
Comment (carlov20) [reply .circleci/config.yml:56]: I've pulled this from the wrong place, in the web module we will need to check for outdated govuk frontend like the template does https://github.com/ministryofjustice/hmpps-template-typescript/blob/main/.circleci/config.yml#L74 minus the typescript check

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+FROM base as build
+ARG BUILD_NUMBER
+ARG GIT_REF
+
+RUN apt-get install -y make python g++ git
+
+COPY package*.json ./
+RUN CYPRESS_INSTALL_BINARY=0 npm ci --no-audit
+
+COPY . .
```
[the remark this answers is not in the document]
Comment (carlov20) [reply Dockerfile:30]: we restrict what is finally put into the container https://github.com/ministryofjustice/wmt-web/blob/WFP-322-pipeline/Dockerfile#L47...#L62 to only the essential files, this was copied from the template project and modified for our needs

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**